packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -951,17 +951,23 @@ export const $access: AccessDecorator = (
 
 export function getAccessOverride(
   context: TCGCContext,
-  entity: Model | Enum | Operation | Union
+  entity: Model | Enum | Operation | Union | Namespace
 ): AccessFlags | undefined {
-  return getScopedDecoratorData(context, accessKey, entity);
+  const accessOverride = getScopedDecoratorData(context, accessKey, entity);
+
+  if(!accessOverride && entity.namespace) {
+    return getAccessOverride(context, entity.namespace);
+  }
+
+  return accessOverride;
 }
 
 export function getAccess(
   context: TCGCContext,
-  entity: Model | Enum | Operation | Union
-): AccessFlags {
+  entity: Model | Enum | Operation | Union | Namespace
+) {
```

Comment:
this func is used for get final calculation result of access, so i don't think we should add namespace